CRM/Utils/System/Base.php

```diff
@@ -945,4 +945,24 @@ public function synchronizeUsers() {
     return [];
   }
 
+  /**
+   * Send an HTTP Response base on PSR HTTP RespnseInterface response.
+   *
+   * @param \Psr\Http\Message\ResponseInterface $response
+   */
+  public function sendResponse(\Psr\Http\Message\ResponseInterface $response) {
+    if (function_exists('http_response_code')) {
+      // PHP 5.4+
+      http_response_code($response->getStatusCode());
+    }
+    else {
+      // @todo should we remove this given we are no longer supporting < PHP 5.6 however keeping
+      // in for the moment for compatability with the original AssetBuilder code.
+      header('X-PHP-Response-Code: ' . $response->getStatusCode(), TRUE, $reponse->getStatusCode());
+    }
+    CRM_Utils_System::setHttpHeader('Content-Type', $response->getHeaderLine('Content-Type'));
```

Comment:
Hrm, shouldn't this be more like:

```php

foreach ($message->getHeaders() as $name => $values) {
  foreach ($values as $value) {
    CRM_Utils_System::setHttpHeader($name, $value);
   }
}

// or

foreach ($response->getHeaders() as $name => $values) {
    CRM_Utils_System::setHttpHeader($name, implode(', ', $values));
}
```

(Riffing on [the docblock for `MessageInterface::getHeaders()`](https://www.php-fig.org/psr/psr-7/#31-psrhttpmessagemessageinterface) -- but I'm not sure which formulation is better here.)